src/jit/flowgraph.cpp

```diff
@@ -4225,51 +4225,55 @@ void                Compiler::fgMarkJumpTarget(BYTE* jumpTarget, unsigned offs)
     jumpTarget[offs] |= (jumpTarget[offs] & JT_JUMP) << 1 | JT_JUMP;
 }
 
-/*****************************************************************************
- *
- *  Walk the instrs and for any jumps we find set the appropriate entry
- *  in the 'jumpTarget' table.
- *  Also sets lvAddrExposed and lvArgWrite in lvaTable[]
- */
-
-#ifdef _PREFAST_
-#pragma warning(push)
-#pragma warning(disable:21000) // Suppress PREFast warning about overly large function
-#endif
-
-//Helpers to track stack transitions for conditional heuristics in fgFindJumpTargets
-const unsigned fgStackSlotInvalid = UINT_MAX;
-const unsigned fgStackSlotUnknown = 0;
-const unsigned fgStackSlotConstant = 1;
-const unsigned fgStackSlotArrayLen = 2;
-const unsigned fgStackSlotFirstArg = 3;
+//------------------------------------------------------------------------
+// FgStack: simple stack model for the inlinee's evaluation stack.
+//
+// Model the inputs available to various operations in the inline body.
+// Tracks constants, arguments, array lengths.
 
-class fgStack
+class FgStack
 {
 public:
-    fgStack() : depth(0)
-    {
-        slot0 = fgStackSlotInvalid;
-        slot1 = fgStackSlotInvalid;
-    }
-
-    //Removes everything from the stack.
-    inline void clear() { depth = 0; }
-    inline void pushUnknown() { push(fgStackSlotUnknown); }
-    inline void pushConstant() { push(fgStackSlotConstant); }
-    inline void pushArrayLen() { push(fgStackSlotArrayLen); }
-    inline void pushArgument(unsigned arg) { push(fgStackSlotFirstArg + arg); }
-    inline unsigned getSlot0() { assert(depth >= 1); return slot0; }
-    inline unsigned getSlot1() { assert(depth >= 2); return slot1; }
-    inline static bool isConstant(unsigned value) { return value == fgStackSlotConstant;}
-    inline static bool isArrayLen(unsigned value) { return value == fgStackSlotArrayLen;}
-    inline static bool isArgument(unsigned value) { return value >= fgStackSlotFirstArg;}
-    inline static unsigned slotTypeToArgNum(unsigned value) { return value - fgStackSlotFirstArg; }
-    inline bool isStackTwoDeep() { return depth == 2; }
-    inline bool isStackOneDeep() { return depth == 1; }
+
+    FgStack()
+        : depth(0)
```

Comment:
Clang is complaining about the initialization order here:
```
/mnt/resource/j/workspace/dotnet_coreclr/master/debug_centos7.1_prtest/src/jit/flowgraph.cpp:4239:11: error: field 'depth' will be initialized after field 'slot0' [-Werror,-Wreorder]
09:45:12         : depth(0)
09:45:12           ^
```

Looks like `depth` needs to be initialized last.